module tls give exception on error but don't write the result as in zgrab

I have done a large scan on my network for port 443, not all host which have open port 443 has a web server HTTPS, so zgrab2 give me an error with stack trace but don't log the event in output file.
zgrab instead report all errors in output file, very useful for post-processing.

    echo IP_WITH_443_OPEN_PORT_BUT_NO_HTTPS | /opt/scanner/sbin/zgrab2 tls --heartbleed --output-file=xxxxx.json
    INFO[0000] started grab at 2018-03-26T09:39:30+02:00    
    DEBU[0003] Failed to detect error from tls: oversized record received with length 20527 at goroutine 894 [running]:
    runtime/debug.Stack(0xd9e8e0, 0xc420366d90, 0xc42003a040)
    	/usr/lib/go-1.9/src/runtime/debug/stack.go:24 +0xa7
    github.com/zmap/zgrab2.TryGetScanStatus(0xd9e8e0, 0xc420366d90, 0xc420366d90, 0xc420358e00)
    	/root/go/src/github.com/zmap/zgrab2/status.go:92 +0x2c0
    github.com/zmap/zgrab2/modules.(*TLSScanner).Scan(0xc42000e528, 0xc420368320, 0x10, 0x10, 0x0, 0x0, 0xa19060, 0x1, 0x4, 0x7f929d28acd0, ...)
    	/root/go/src/github.com/zmap/zgrab2/modules/tls.go:72 +0xec
    github.com/zmap/zgrab2.RunScanner(0xda9600, 0xc42000e528, 0xc420366d20, 0xc420368320, 0x10, 0x10, 0x0, 0x0, 0xa4a144, 0x3, ...)
    	/root/go/src/github.com/zmap/zgrab2/scanner.go:32 +0xcc
    github.com/zmap/zgrab2.grabTarget(0xc420368320, 0x10, 0x10, 0x0, 0x0, 0xc420366d20, 0x0, 0x0, 0x0)
    	/root/go/src/github.com/zmap/zgrab2/processing.go:91 +0x245
    github.com/zmap/zgrab2.Process.func2(0xc420057b00, 0xc420366d20, 0xc420057b60, 0xc4203682e0, 0x374)
            /root/go/src/github.com/zmap/zgrab2/processing.go:150 +0x143
    created by github.com/zmap/zgrab2.Process
    	/root/go/src/github.com/zmap/zgrab2/processing.go:143 +0x191

output file xxxx.json is empty
with zgrab instead

    echo IP_WITH_443_OPEN_PORT_BUT_NO_HTTPS | /opt/scanner/sbin/zgrab --port 443 --tls --heartbleed --output-file=xxxxx.json
    Mar 26 09:39:50.173 [INFO] banner-grab: started grab at 2018-03-26T09:39:50+02:00
    Mar 26 09:39:52.851 [ERROR] banner-grab: Conversation error with remote host 2.238.76.236:443: tls: oversized record received with length 20527
    Mar 26 09:39:54.607 [INFO] banner-grab: finished grab (0 success; 1 failure) at 2018-03-26T09:39:54+02:00

output file xxxxx.json 

    cat xxxxx.json 
    {"ip":"IP_WITH_443_OPEN_PORT_BUT_NO_HTTPS","timestamp":"2018-03-26T09:39:52+02:00","data":{"tls":{}},"error":"tls: oversized record received with length 20527","error_component":"tls"}

I have used golang 1.10 for tests.

--- a/status.go
+++ b/status.go
@@ -1,6 +1,7 @@
 package zgrab2
 
 import (
+	"errors"
 	"io"
 	"net"
 	"runtime/debug"
@@ -67,11 +68,13 @@ func TryGetScanStatus(err error) ScanStatus {
 		// Presumably the caller did not call TryGetScanStatus if the EOF was expected
 		return SCAN_IO_TIMEOUT
 	}
-	switch e := err.(type) {
-	case *ScanError:
-		return e.Status
-	case *net.OpError:
-		switch e.Op {
+	var scanError *ScanError
+	var opError *net.OpError
+	switch {
+	case errors.As(err, &scanError):
+		return scanError.Status
+	case errors.As(err, &opError):
+		switch opError.Op {
 		case "dial":
 			// TODO: Distinguish connection timeout / connection refused
 			// Windows examples:
@@ -86,12 +89,12 @@ func TryGetScanStatus(err error) ScanStatus {
 			return SCAN_IO_TIMEOUT
 		default:
 			// TODO: Do we need a generic network error?
-			log.Debugf("Failed to detect error from net.OpError %v, op = %s at %s", e, e.Op, string(debug.Stack()))
+			log.Debugf("Failed to detect error from net.OpError %v, op = %s at %s", opError, opError.Op, string(debug.Stack()))
 			return SCAN_UNKNOWN_ERROR
 		}
 	// TODO: More error types
 	default:
-		log.Debugf("Failed to detect error from %v at %s", e, string(debug.Stack()))
+		log.Debugf("Failed to detect error from %v at %s", err, string(debug.Stack()))
 		return SCAN_UNKNOWN_ERROR
 	}
 }

